File: examples/scala/src/main/scala/com/xebia/functional/xef/scala/auto/Population.scala
package com.xebia.functional.xef.scala.conversation

import com.xebia.functional.xef.scala.conversation.*
import com.xebia.functional.xef.llm.models.images.ImagesGenerationResponse
import io.circe.Decoder
import com.xebia.functional.xef.prompt.Prompt

private final case class Population(size: Int, description: String) derives SerialDescriptor, Decoder

private final case class Image(description: String, url: String) derives SerialDescriptor, Decoder

@main def runPopulation: Unit =
  conversation {
    val cadiz: Population = prompt(Prompt("Population of Cádiz, Spain."))
    val seattle: Population = prompt(Prompt("Population of Seattle, WA."))
    val imgs: ImagesGenerationResponse = images(Prompt("A hybrid city of Cádiz, Spain and Seattle, US."))
    println(imgs)
    println(s"The population of Cádiz is ${cadiz.size} and the population of Seattle is ${seattle.size}")
  }

File: examples/kotlin/src/main/kotlin/com/xebia/functional/xef/auto/Population.kt
package com.xebia.functional.xef.auto

import com.xebia.functional.xef.auto.llm.openai.conversation
import com.xebia.functional.xef.auto.llm.openai.image
import kotlinx.serialization.Serializable

@Serializable
data class Population(val size: Int, val description: String)

@Serializable
data class Image(
    val description: String,
    val url: String,
)

suspend fun main() =
    conversation {
        val img: Image = image("")
        println(img)
    }
